x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -92,8 +97,11 @@ public class TransportGetStackTracesAction extends HandledTransportAction<GetSta
         Setting.Property.NodeScope
     );
 
+    private static final int maxSupportedUniqueHosts = 10_000;
```

Comment:
IMHO that's good enough for now but I wonder whether we should turn this into a setting that we can tweak (if yes, then we can do this in a follow-up PR at a later time). This can be useful e.g. in support cases. To be clear. I'm *not* proposing that this should be an official and documented setting but rather an escape hatch that we can use if we need to without requiring a new release.

Reply:
Made this the 150k constant now, see the comment/explanation why this won't explode.
But we should chat about it and possible side effects anyway.